sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -59,6 +59,12 @@ def __init__(self, http_client, properties, ipython_display,
         super(LivySession, self).__init__()
         assert u"kind" in list(properties.keys())
         kind = properties[u"kind"]
+
+        if not should_heartbeat:
+            properties.pop(u"heartbeatTimeoutInSecond")
+        elif properties[u"heartbeatTimeoutInSecond"] == 0:
+            properties.pop(u"heartbeatTimeoutInSecond")
```

Comment:
I deliberately skipped that for backward compat reason. For this version of magic to work with Livy 0.2, user can turn on heartbeat but set timeout to 0. That way, we will still start the heartbeat thread in jupyter but not send the extra param to livy (which would cause 0.2 livy to fail).